src/App.vue

```diff
@@ -95,35 +107,73 @@ export default defineComponent({
         },
         completed(): Todo[] {
             return this.todos.filter(todo => todo.isCompleted);
+        },
+        isAppReady(): boolean {
+            return !this.isLoadingUserSession && !this.isLoadingTodos;
+        },
+        isLoggedIn(): boolean {
+            return Boolean(this.user.uid);
+        },
+        userTodosPath(): string {
+            return `users/${this.user.uid}/todos`;
         }
     },
+    created() {
+        this.setUser();
+    },
     methods: {
+        setUser() {
+            firebaseApp.auth().onAuthStateChanged((user: User) => {
+                this.isLoadingUserSession = false;
+
+                if (user) {
+                    this.user = user;
+                    this.getUserTodos();
+                } else {
+                    this.isLoadingTodos = false;
+                }
+            })
+        },
+        getUserTodos() {
+            db.ref(this.userTodosPath).on("value", (snap: any) => {
+                this.todos = Object.values(snap.val());
+                this.isLoadingTodos = false;
+            });
+        },
         getRand(): string {
             return new Date().getTime().toString() + Math.floor(Math.random() * 1000000);
         },
         toggleCompleted(todoId: string, value: boolean): void {
-            todosRef.child(todoId).update({ isCompleted: value })
+            db.ref(this.userTodosPath).child(todoId).update({ isCompleted: value })
         },
         addTodo(): void {
-            if (!this.newTodo) {
-                return;
-            }
-            const newTodo: Todo = {
-                isCompleted: false,
-                title: this.newTodo,
-                id: `todo-${this.getRand()}`
+            if (this.newTodo) {
+                const newTodo: Todo = {
+                    isCompleted: false,
+                    title: this.newTodo,
+                    id: `todo-${this.getRand()}`
+                }
+                db.ref(this.userTodosPath).child(newTodo.id).set(newTodo)
+                this.newTodo = "";
             }
-            todosRef.child(newTodo.id).set(newTodo);
             this.newTodo = "";
         },
-        editTodo(todoId: string, title: string): void {
-            todosRef.child(todoId).update({ title });
+        editTodo(todoId: string, value: string): void {
+            db.ref(this.userTodosPath).child(todoId).update({ title: value })
```

Comment:
Question: Why did you change back title -> value?

Reply:
merging conflicts mistake, gotta change it back